plugin/src/main/kotlin/com/testknight/startup/WelcomePage.kt

```diff
@@ -1,12 +1,109 @@
 package com.testknight.startup
 
+import com.intellij.ide.BrowserUtil
+import com.intellij.ide.plugins.IdeaPluginDescriptor
 import com.intellij.ide.plugins.PluginInstaller
+import com.intellij.ide.plugins.PluginStateListener
+import com.intellij.ide.util.PropertiesComponent
+import com.intellij.ide.util.RunOnceUtil
+import com.intellij.openapi.application.ApplicationManager
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.startup.StartupActivity
-import com.testknight.listeners.WelcomeListener
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.ui.components.JBCheckBox
+import com.intellij.ui.layout.panel
+import com.testknight.settings.SettingsService
+import javax.swing.Action
+import javax.swing.JComponent
 
 class WelcomePage : StartupActivity {
+
+    private var settingsState = SettingsService.instance.state
+
     override fun runActivity(project: Project) {
-        PluginInstaller.addStateListener(WelcomeListener(project))
+
+        PluginInstaller.addStateListener(UninstallListener())
+
+        if (!ApplicationManager.getApplication().isHeadlessEnvironment) {
+            RunOnceUtil.runOnceForApp(
+                "HasShownTestKnightIntro"
```

Comment:
This seems like a magic string to me. What is its purpose?

Reply:
The purpose of this string is to give a sort of an "id" to the install page. Thus, I used this string for resetting the settings, because maybe at some point the user wants to uninstall the plugin.